pkg/manifests/manifests.go

```diff
@@ -2878,6 +2878,39 @@ func (f *Factory) ControlPlanePrometheusRule() (*monv1.PrometheusRule, error) {
 		}
 		r.Spec.Groups = groups
 	}
+	if !f.infrastructure.HighlyAvailableInfrastructure() {
+		// We are in SNO. Aggregated API may go down at times since there is no high
+		// availability for these components. Examples of these situations may
+		// include upgrades or certificate rotations. To try to avoid expected alerts
+		// we change the KubeAggregatedAPIDown severity to info, as its not an anomaly,
+		// and introduce a new alert KubeAggregatedAPIOffline, which fires over longer
+		// window times to detect downtime when its not expected.
+		for i := range r.Spec.Groups {
+			group := &r.Spec.Groups[i]
+			if group.Name == "kubernetes-system-apiserver" {
+				for j := range group.Rules {
+					rule := &group.Rules[j]
+					if rule.Alert == "KubeAggregatedAPIDown" {
+						rule.Labels["severity"] = "info"
+						break
+					}
+				}
+				group.Rules = append(group.Rules, monv1.Rule{
+					Alert: "KubeAggregatedAPIUnavailable",
+					Annotations: map[string]string{
+						"description": `Kubernetes aggregated API {{ $labels.name }}
+/{{ $labels.namespace }} has been only {{ $value | humanize }}% available over the
+last 20m.`},
+					Expr: intstr.FromString("(1 - max by(name, namespace)(avg_over_time(aggregator_unavailable_apiservice[20m]))) * 100 < 85"),
```

Comment:
When prometheus first starts collecting metrics on SNO, it sometimes begins while some of the aggregated APIs are still down for a brief period. So the metrics look something like this:

```
<blank>-<blank>-<blank>-<blank>-<blank>-<blank>-<blank>-<blank>-<down>-<down>-<up>-<up>-<up>-...
```

If:
1) blanks are considered "down" by this expression, then we're surely gonna trigger this alert
2) blanks are ignored by this expression and the window is just cut short, then we're also surely gonna trigger this alert because in the first 5m the window is growing from 0m to 5m and the spike of <down>s we have at the beginning is likely gonna make up a >85% portion of it

In case of (1) -
Is there anything we can do in this query to pretend that `<blank>`s are actually `<up>`?

In case of (2) - Can we make sure that the expression is only true when there's actually enough data to fill a 20m window?